examples/jsm/exporters/GLTFExporter.js

```diff
@@ -602,16 +602,27 @@ class GLTFWriter {
 	}
 
 	/**
-	 * Assign and return a temporal unique id for an object
-	 * especially which doesn't have .uuid
+	 * Returns ids for buffer attributes.
 	 * @param  {Object} object
 	 * @return {Integer}
 	 */
-	getUID( object ) {
+	getUID( attribute, morphTargets = null ) {
 
-		if ( ! this.uids.has( object ) ) this.uids.set( object, this.uid ++ );
+		if ( this.uids.has( attribute ) === false ) {
 
-		return this.uids.get( object );
+			const uids = new Map();
+
+			uids.set( true, this.uid ++ );
+			uids.set( false, this.uid ++ );
+
+			this.uids.set( attribute, uids );
+
+		}
+
+		const uids = this.uids.get( attribute );
+		const hasMorphTargets = Array.isArray( morphTargets ) && morphTargets.length > 0;
+
+		return uids.get( hasMorphTargets );
```

Comment:
I'm having a difficult time understanding this method – a few questions:

- Why `uids.set( true, this.uid ++ );` and `uids.set( false, this.uid ++ );`, What do those lines do?
- Since we don't seem to use the morph target object for anything more than a boolean check, could we replace it with a boolean? Possibly `isRelativeCopy` or something like that. 

Note that if the attribute were used both for the base geometry and a morph target, and the morph target were _relative_, we wouldn't need to copy it or do any of this. That's very unlikely (how could the same attribute be used for both relative and absolute data?) and so I'm not really worried about duplicate data in that case, but just to say that it's not the fact that the attribute "hasMorphTargets" that we actually care about here, it's the fact that we cloned the original and rewrote it as relative deltas.

Now that three.js supports relative morph targets, we could even put this in an external utility function.